README.md

```diff
@@ -85,11 +85,48 @@ To run specific test suites, use the `testSuites` package script:
  yarn run testSuites -n polygon basic
 ```
 
-# Deployment
+# Deploying on Mangrove
 
-## FIXME
+Mangrove uses `forge script`
 
-No deployment story with foundry yet -- but foundry has the features, we just need to write the deployments and the tooling around it.
+[`forge script <scriptName>`](https://book.getfoundry.sh/reference/forge/forge-script) executes an arbitrary smart contract function locally. Then, any CALLs executed therein and preceded by the cheatcode [`vm.broadcast()`](https://book.getfoundry.sh/cheatcodes/broadcast) can be broadcast to a remote node by reading a forge-generated `run-*.json` field.
+
+## Generating Mangrove address files
+
+The log of transactions generated by `forge script` gets written to `broadcast/<scriptName>/<chainId>/run-latest.json`. It is an array of low-level transactions info with some additions like newly created contract names.
+
+Mangrove needs to:
+
+- Name its contract (multiple instances of the same contract may be deployed)
+- Ignore script names (different scripts are used for different networks)
+- Have all its deployed contracts in one place
+
+To do the above, Mangrove adds a layer to `forge script` deployment.
+
+- Deployment scripts should inherit the `Deployer` contract.
+- You should call `outputDeployment()` at the end of your scripts.
+  - When `outputDeployment()` gets called, a file all known deployed contracts are written to `addresses/deployed.backup/<network>-latest.json`
+- You should set `WRITE_DEPLOY=true` when running scripts that you want to broadcast.
+  - When `WRITE_DEPLOY=true`, the contract set is also written to `addresses/deployed/<network>.json`.
+
+(Note that for mumbai, `network=maticmum`)
```

Comment:
why not mumbai? :)

Reply:
Because ethers calls it maticmum :( so it's consistent all the way to mangrove.js.